system-tests/test/pause_headed_exit_spec.ts

```diff
@@ -0,0 +1,51 @@
+import systemTests from '../lib/system-tests'
+
+describe('cy.pause() works with --headed and --no-exit', () => {
+  systemTests.setup()
+
+  systemTests.it('pauses with --headed and --no-exit', {
+    spec: 'pause_spec.js',
+    snapshot: true,
+    headed: true,
+    noExit: true,
+    expectedExitCode: null,
+    onSpawn: (cp) => {
+      cp.stdout.on('data', (buf) => {
+        if (buf.toString().includes('not exiting due to options.exit being false')) {
+          cp.kill()
+        }
+      })
+    },
+  })
+
+  systemTests.it('does not pause if headless', {
+    spec: 'no_pause_spec.js',
+    snapshot: true,
+    headed: false,
+    noExit: true,
+    expectedExitCode: null,
```

Comment:
what's the exit code? is it...

```suggestion
    expectedExitCode: 1,
```

...? we should get the same exit code every time, so might as well assert on it

Reply:
@flotwig The exit code is always null because it kills the process before returning an exit code. Is there a way to get around this? I tried sending an exit code before/after killing the process but it continued to be null every time.

Reply:
oh yeah, I forgot about this. Node.js actually passes 2 arguments to the `.on('exit')` handler: `exitCode`, AND `signal`. if the process exited from a signal, `exitCode` is null: https://stackoverflow.com/a/39169784/3474615

this is in contrast to how POSIX-compatible shells work, which assign an exit code > 128 if a process exited due to an unhandled signal: https://unix.stackexchange.com/a/99134/305482

so yeah, nothing to do here, this is correct :smile: sorry